### PR TITLE
replaced existing BookRepository with one that uses JpaRepository, ad…

### PR DESCRIPTION
**Pull request by YuraKhilko**
…ded Liquibase support, implemented soft delete concept for Book, added following options with implementations to BookController: update, delete, search.

### CODE REVIEW

File: src/main/java/bookstore/repository/book/spec/CoverImageSpecificationProvider.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package bookstore.repository.book.spec;
+
+import bookstore.model.Book;
+import bookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CoverImageSpecificationProvider implements SpecificationProvider<Book> {
```
Comment (Rommelua) [inline src/main/java/bookstore/repository/book/spec/CoverImageSpecificationProvider.java]: search by CoverImage? I can not imagine how it works

File: src/main/java/bookstore/repository/book/spec/DescriptionSpecificationProvider.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package bookstore.repository.book.spec;
+
+import bookstore.model.Book;
+import bookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DescriptionSpecificationProvider implements SpecificationProvider<Book> {
```
Comment (Rommelua) [inline src/main/java/bookstore/repository/book/spec/DescriptionSpecificationProvider.java]: same, how it works? search by full Description?

File: src/main/java/bookstore/repository/book/spec/PriceSpecificationProvider.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package bookstore.repository.book.spec;
+
+import bookstore.model.Book;
+import bookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PriceSpecificationProvider implements SpecificationProvider<Book> {
```
Comment (Rommelua) [inline src/main/java/bookstore/repository/book/spec/PriceSpecificationProvider.java:9]: Same. We need price bounds for searchI think (min/max)

File: src/main/java/bookstore/repository/book/spec/TitleSpecificationProvider.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package bookstore.repository.book.spec;
+
+import bookstore.model.Book;
+import bookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
```
Comment (Rommelua) [inline src/main/java/bookstore/repository/book/spec/TitleSpecificationProvider.java:9]: let's use "like" logic for Title
for example when we have "master" in parameter we would like to see both
**Master and Margarita** and **Master Your Emotions** in the results

File: src/main/java/bookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -32,7 +36,38 @@ public List<BookDto> findAll() {
...
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = findBookById(id);
+
+        book.setPrice(requestDto.getPrice());
+        book.setIsbn(requestDto.getIsbn());
+        book.setAuthor(requestDto.getAuthor());
+        book.setTitle(requestDto.getTitle());
+        book.setDescription(requestDto.getDescription());
+        book.setCoverImage(requestDto.getCoverImage());
```
Comment (Rommelua) [inline src/main/java/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
        Book book = bookMapper.toModel(bookDto);
        book.setId(id);
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/bookstore/repository/book/spec/CoverImageSpecificationProvider.java (hunk 1 above)
Comment (YuraKhilko) [reply src/main/java/bookstore/repository/book/spec/CoverImageSpecificationProvider.java]: But Google has such option (search by image) :) 
Ok, I've deleted it.

File: src/main/java/bookstore/repository/book/spec/DescriptionSpecificationProvider.java (hunk 2 above)
Comment (YuraKhilko) [reply src/main/java/bookstore/repository/book/spec/DescriptionSpecificationProvider.java]: The main problem - we don't have a TS. So only imagination can stop us.
Ok, I've deleted it.

**Review by aaananas — APPROVED** [review]

**Pull request merged**